enhancements/sig-architecture/11-cluster-deletion/README.md

```diff
@@ -0,0 +1,86 @@
+
+## Summary
+
+Currently, when we remove a cluster from the Hub, the `managedClusterAddons` and `manifestWorks` in the cluster namespace need to be deleted manually, and cannot ensure that the resources deployed on the Spoke cluster can be cleaned up. This proposal is to define clearly what happened when a cluster is deleted.
+
+## Motivation
+
+### Goals
+
+- Define which resources related the deleted cluster should be cleaned up by OCM.
+- Clarify the order in which these resources are deleted.
+
+### Non-Goals
+
+- Clean up customized resources automatically.
+
+## Proposal
+
+### Resources related with managed cluster
+
+On the Hub cluster
+
+1. `managedCluster`. The deletion process should be triggered by deleting `managedCluster`.
+2. `managedClusterAddons` and `manifestWorks`, which should be deleted after the `manangedCluster` is deleted. Because they will not work after the rbac of registration-agent and work-agent is removed.
+3. The other resources in the cluster namespace except `managedClusterAddons` and `manifestWorks` should be cleaned up by their owners.
+4. The cluster namespace, which should be deleted optionally. Because the user wants to reserve some resources deployed by themselves in the namespace after the cluster is deleted in some use cases.
+   
+On the Spoke cluster
+
+1. The resources deployed by the `manifestWorks` should be cleaned up.
+2. The resources deployed by other owners can be cleaned up optionally by their owners like some application resources.
+3. The `Klusterlet` CR should be deleted.
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, all cluster resources should be cleaned up after I delete the `managedCluster`,including the cluster namespace on the hub cluster and the resources deployed by `manifestWork` on the spoke cluster.
+
+#### Story 2
+
+As a cluster admin, the cluster namespace and some specified `manifestWorks` should not be deleted after I delete the `managedCluster`, because I want to delete some resources in the cluster namespace by myself.
+
+## Design Details 
+
+1. Add a `deleteOption` filed in the spec of `managedCluster`. 
+   1. With the value `deleteAll`, means to delete the cluster namespace after the `managedCluster` is deleted. 
+   2. Otherwise, the cluster namespace will not be deleted by default.
+2. Add a `phase` filed in the status of `managedCluster`, which is the current lifecycle phase of `managedCluster`. 
+   1. `Pending` phase means the `managedCluster` is waiting to be joined.
+   2. `Joined` phase means the `managedCluster` is joined.
+   3. `Terminating` phase means the `Terminating` is being deleted.
+3. Define an annotation `open-cluster-management.io/delete-by-user`. The `managedClusterAddons` and `manifestWorks` with this annotation will not be deleted by OCM during the deletion process.
+
+### The behaviors when a cluster is removed from the Hub.
+
+The cluster admin deletes the `managedCluster` on the Hub.
```

Comment:
we should consider two cases
1. user delete cluster at first on hub.
2. user remove klusterlet at first on managed cluster.

Reply:
> 2. user remove klusterlet at first on managed cluster.

if the managedCluster is not deleted after user removed klusterlet,  we should do nothing.
I think we should consider 1 case that the managedCluster is deleted or not.

Reply:
you probably need to consider a grace period on hub when klusterlet is removed already, then do a force delete?